fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/BaseMetadataAvailability.java

```diff
@@ -0,0 +1,56 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.table.availability;
+
+import com.yahoo.bard.webservice.data.config.names.DataSourceName;
+import com.yahoo.bard.webservice.metadata.DataSourceMetadataService;
+import com.yahoo.bard.webservice.table.resolver.PhysicalDataSourceConstraint;
+import com.yahoo.bard.webservice.util.SimplifiedIntervalList;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+
+import javax.validation.constraints.NotNull;
+
+/**
+ * An availability based on a DataSourceMetadataService, which is backed by a single data source in that service.
+ */
+public abstract class BaseMetadataAvailability implements Availability {
```

Comment:
It's more than that it has only 1 data source. In particular, it _must_ have a `metadataService` (as indicated in the javadoc), where the composite ones do not have one. The 1 datasource aspect is really ancillary, and not the main purpose, though it does tend to go hand-in-hand.

@michael-mclawhorn do you have thoughts here? or anyone else? I'm happy to change it, but it's more than just "not composite".